### ISSUE TITLE
Failed to register 404 not found

### ISSUE THREAD

**Issue by jprates** [haberda/hassio_addons#74]
As discussed on [this other issue](https://github.com/bbernhard/signal-cli-rest-api/issues/384) please be so kind as to release a new version of hass signal messenger add-on using @bbernhard's signal-cli-rest-api latest version which is using signal-cli v0.11.11.

Thanks.

**Comment by jprates** [comment]
This one is revealing itself as a tough bug to kill.

Thanks a lot for the new release @haberda , but the bug is still there.

From hass shell I even tried with 3 ips, the loopback, the one from the HASS itself, and the one that appears on the add-on log as the HOST_IP. All failed with same 404 error.

```
➜  ~ curl -X POST -H "Content-Type: application/json" --data '{"use_voice": false}' 'http://127.0.0.1:8080/v1/register/+351910446029'
{"error":"Failed to register: [404] Not found\n"}#
➜  ~ curl -X POST -H "Content-Type: application/json" --data '{"use_voice": false}' 'http://172.30.33.0:8080/v1/register/+351910446029'
{"error":"Failed to register: [404] Not found\n"}#
➜  ~ curl -X POST -H "Content-Type: application/json" --data '{"use_voice": false}' 'http://192.168.1.70:8080/v1/register/+351910446029'
{"error":"Failed to register: [404] Not found\n"}#
➜  ~ docker exec -it 6b5fccc762b7 /bin/bash
WARNING: signal-cli-rest-api runs as signal-api (not as root!)
Run 'su signal-api' before using signal-cli!
If you want to use signal-cli directly, don't forget to specify the config directory. e.g: "signal-cli --config /data"
root@1315902c-signal-messenger:/data/data# su signal-api
signal-api@1315902c-signal-messenger:/data/data$ signal-cli --config /data -a +351910446029 register
Failed to register: [404] Not found
```

Let me know if I can assist with some more info from my side.

It would be great to know if when you try to register a new number on your test env you succeed or if you also fail.

tia
-jprates

**Comment by haberda** [comment]
Registration works fine for me using both pathways. Have you tried a fresh install? Your config might be corrupted.

**Comment by amroncz** [comment]
> Thanks a lot for the new release @haberda , but the bug is still there.
Can confirm that.

> Have you tried a fresh install?
I tried a fresh install and loopback address and machine address

**Comment by jprates** [comment]
Yup, just did a fresh install and restart to HASS, still the same.

Guide us through to what tests and/or logs you need so that we can pinpoint what's going on to assist debugging this.

**Comment by haberda** [comment]
Make sure you are trying to register in normal mode. The only other thing I can suggest is make sure the container has internet access.

What does the log look like when the container starts?

**Comment by jprates** [comment]
```
+ set -e
+ [ -z /data ]
+ usermod -u 1000 signal-api
usermod: no changes
+ groupmod -g 1000 signal-api
+ chown 1000:1000 -R /data
+ cat
+ cap_prefix=-cap_
+ cat /proc/sys/kernel/cap_last_cap
+ seq -s ,-cap_ 0 40
+ caps=-cap_0,-cap_1,-cap_2,-cap_3,-cap_4,-cap_5,-cap_6,-cap_7,-cap_8,-cap_9,-cap_10,-cap_11,-cap_12,-cap_13,-cap_14,-cap_15,-cap_16,-cap_17,-cap_18,-cap_19,-cap_20,-cap_21,-cap_22,-cap_23,-cap_24,-cap_25,-cap_26,-cap_27,-cap_28,-cap_29,-cap_30,-cap_31,-cap_32,-cap_33,-cap_34,-cap_35,-cap_36,-cap_37,-cap_38,-cap_39,-cap_40
+ [ normal = json-rpc ]
+ hostname+ awk -I
 {print $1}
+ export HOST_IP=172.30.33.0
+ exec setpriv --reuid=1000 --regid=1000 --init-groups --inh-caps=-cap_0,-cap_1,-cap_2,-cap_3,-cap_4,-cap_5,-cap_6,-cap_7,-cap_8,-cap_9,-cap_10,-cap_11,-cap_12,-cap_13,-cap_14,-cap_15,-cap_16,-cap_17,-cap_18,-cap_19,-cap_20,-cap_21,-cap_22,-cap_23,-cap_24,-cap_25,-cap_26,-cap_27,-cap_28,-cap_29,-cap_30,-cap_31,-cap_32,-cap_33,-cap_34,-cap_35,-cap_36,-cap_37,-cap_38,-cap_39,-cap_40 signal-cli-rest-api -signal-cli-config=/data
time="2023-07-16T20:43:00+01:00" level=info msg="Started Signal Messenger REST API"
[GIN] 2023/07/16 - 20:44:38 | 400 | 23.213715118s |     172.30.32.1 | POST     "/v1/register/+351xxxxxxxxx"
```

Any idea on how to test internet connection from inside your container? I'm out of options here:
```
root@1315902c-signal-messenger:/var/log# which nslookup
root@1315902c-signal-messenger:/var/log# which ping
root@1315902c-signal-messenger:/var/log# which dig
root@1315902c-signal-messenger:/var/log# which host

```

It seems it lacks ping and telnet, and having this nameserver entry is suspicious to me, what is 127.0.0.11?:

```
root@1315902c-signal-messenger:/var/log# cat /etc/resolv.conf
search local.hass.io
nameserver 127.0.0.11
options ndots:0

```

**Comment by jprates** [comment]
update: I went ahead and installed ping inside your container:

apt-get install iputils-ping -y

I can now ping several public ip addresses, so internet access is not an issue.

**Comment by bbernhard** [comment]
If you register the number directly via signal-cli (see https://github.com/bbernhard/signal-cli-rest-api/issues/384#issuecomment-1634905945) does it work then?

**Comment by jprates** [comment]
@bbernhard you can see in the original entry I've done that with the same results: https://github.com/haberda/hassio_addons/issues/74#issuecomment-1637150295

I believe this is what you ask for, right?

```
signal-api@1315902c-signal-messenger:/var/log$ signal-cli --config /data -a +351xxxxxxxxx register
Failed to register: [404] Not found
```

**Comment by amroncz** [comment]
Interesting:
When I try to UNregister a number, as expected, I get a message that the number is not registered.
So is it a specific problem with registering or am I missing something?

**Comment by jprates** [comment]
here you go, as requested:

```
signal-api@1315902c-signal-messenger:/data/data$ signal-cli --verbose --config /data -a +351xxxxxxxxx register
2023-07-17T17:02:05.713+0100 [main] DEBUG org.asamk.signal.util.IOUtils - XDG_RUNTIME_DIR not set, falling back to temp dir
2023-07-17T17:02:06.798+0100 [main] INFO  LibSignal - [libsignal]: rust/bridge/jni/src/logging.rs:156: Initializing libsignal version:0.22.0
2023-07-17T17:02:09.327+0100 [main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Starting...
2023-07-17T17:02:09.953+0100 [main] INFO  com.zaxxer.hikari.pool.HikariPool - HikariPool-1 - Added connection org.sqlite.jdbc4.JDBC4Connection@73163d48
2023-07-17T17:02:09.966+0100 [main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Start completed.
2023-07-17T17:02:14.299+0100 [main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Shutdown initiated...
2023-07-17T17:02:14.314+0100 [main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Shutdown completed.
Failed to register: [404] Not found
org.whispersystems.signalservice.api.push.exceptions.NotFoundException: [404] Not found
        at org.whispersystems.signalservice.internal.push.PushServiceSocket.validateServiceResponse(PushServiceSocket.java:1741)
        at org.whispersystems.signalservice.internal.push.PushServiceSocket.makeServiceRequest(PushServiceSocket.java:1722)
        at org.whispersystems.signalservice.internal.push.PushServiceSocket.makeServiceRequest(PushServiceSocket.java:1660)
        at org.whispersystems.signalservice.internal.push.PushServiceSocket.requestSmsVerificationCode(PushServiceSocket.java:333)
        at org.whispersystems.signalservice.api.SignalServiceAccountManager.requestSmsVerificationCode(SignalServiceAccountManager.java:240)
        at org.asamk.signal.manager.util.NumberVerificationUtils.requestVerificationCode(NumberVerificationUtils.java:35)
        at org.asamk.signal.manager.RegistrationManagerImpl.register(RegistrationManagerImpl.java:120)
        at org.asamk.signal.commands.RegisterCommand.register(RegisterCommand.java:67)
        at org.asamk.signal.commands.RegisterCommand.handleCommand(RegisterCommand.java:43)
        at org.asamk.signal.App.handleRegistrationCommand(App.java:261)
        at org.asamk.signal.App.init(App.java:221)
        at org.asamk.signal.Main.main(Main.java:60)

```

**Comment by jprates** [comment]
Please note libsignal version:0.22.0 vs version:0.25.0 !

**Comment by bbernhard** [comment]
Are you sure that you are running the latest HASS addon? Could you please do a `signal-cli --version` inside the docker container.

**Comment by jprates** [comment]
Yup.

HASS add-on info:

![imagem](https://github.com/haberda/hassio_addons/assets/20585218/cfb1ab87-68db-41c8-a2f6-12ebf4cfb4cd)

Inside docker container running on HASS:

![imagem](https://github.com/haberda/hassio_addons/assets/20585218/29470d38-9534-4a66-9987-3fc6a03eead9)

**Comment by bbernhard** [comment]
I am not familiar with the HASS addons, but do you maybe need to restart something? It looks like you are running still the old version

**Comment by littlejogi** [comment]
I have the same problem. I think there is a problem with the 0.67er version. It seems it contains v0.11.7 of the signal-cli.

**Comment by jprates** [comment]
When you install a new version a restart is performed automatically.

But sure, no prob, I can do it again... one sec... nope, same results:

```
➜  ~ docker ps | grep signal
00528bf08b2f   ghcr.io/haberda/signal_messenger/amd64:0.67.0              "/options.sh"         43 seconds ago   Up 35 seconds (healthy)   0.0.0.0:8080->8080/tcp, :::8080->8080/tcp             addon_1315902c_signal_messenger
➜  ~ docker exec -it 00528bf08b2f  /bin/bash
WARNING: signal-cli-rest-api runs as signal-api (not as root!)
Run 'su signal-api' before using signal-cli!
If you want to use signal-cli directly, don't forget to specify the config directory. e.g: "signal-cli --config /data"
root@1315902c-signal-messenger:/data/data# uptime
 18:19:07 up 29 days,  4:58,  0 users,  load average: 0.67, 0.59, 0.45
root@1315902c-signal-messenger:/data/data# signal-cli --version
signal-cli 0.11.7
root@1315902c-signal-messenger:/data/data# signal-cli --verbose --config /data -a +351xxxxxxxxx register
2023-07-18T18:19:53.417+0100 [main] DEBUG org.asamk.signal.util.IOUtils - XDG_RUNTIME_DIR not set, falling back to temp dir
2023-07-18T18:19:54.691+0100 [main] INFO  LibSignal - [libsignal]: rust/bridge/jni/src/logging.rs:156: Initializing libsignal version:0.22.0
2023-07-18T18:19:57.777+0100 [main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Starting...
2023-07-18T18:19:58.454+0100 [main] INFO  com.zaxxer.hikari.pool.HikariPool - HikariPool-1 - Added connection org.sqlite.jdbc4.JDBC4Connection@73163d48
2023-07-18T18:19:58.471+0100 [main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Start completed.
2023-07-18T18:20:04.251+0100 [main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Shutdown initiated...
2023-07-18T18:20:04.265+0100 [main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Shutdown completed.
Failed to register: [404] Not found
org.whispersystems.signalservice.api.push.exceptions.NotFoundException: [404] Not found
        at org.whispersystems.signalservice.internal.push.PushServiceSocket.validateServiceResponse(PushServiceSocket.java:1741)
        at org.whispersystems.signalservice.internal.push.PushServiceSocket.makeServiceRequest(PushServiceSocket.java:1722)
        at org.whispersystems.signalservice.internal.push.PushServiceSocket.makeServiceRequest(PushServiceSocket.java:1660)
        at org.whispersystems.signalservice.internal.push.PushServiceSocket.requestSmsVerificationCode(PushServiceSocket.java:333)
        at org.whispersystems.signalservice.api.SignalServiceAccountManager.requestSmsVerificationCode(SignalServiceAccountManager.java:240)
        at org.asamk.signal.manager.util.NumberVerificationUtils.requestVerificationCode(NumberVerificationUtils.java:35)
        at org.asamk.signal.manager.RegistrationManagerImpl.register(RegistrationManagerImpl.java:120)
        at org.asamk.signal.commands.RegisterCommand.register(RegisterCommand.java:67)
        at org.asamk.signal.commands.RegisterCommand.handleCommand(RegisterCommand.java:43)
        at org.asamk.signal.App.handleRegistrationCommand(App.java:261)
        at org.asamk.signal.App.init(App.java:221)
        at org.asamk.signal.Main.main(Main.java:60)
```

**Comment by bsoufflet** [comment]
For info I have the exact same problem with 0.67.0 on a fresh install...

**Comment by jprates** [comment]
Hi @bbernhard , @haberda , is any of you planning to fix this?

Don't want to be a pain, but I'm just wandering if by chance you're not waiting on one another to do something...

This is the only notification tool I use on my system, and my users are complaining a lot already because they are not warned about some important events anymore.

I did my best to aid diagnose the issue, as did others, can you guys please let us know how this issue stands and what's the plan?

Again thank you for your hard (and unpaid) work for us all.

**Comment by haberda** [comment]
I need to do some investigating. As you suggest I work full time and have a family, this is a hobby project at best. I will try and take a look this weekend.

**Comment by haberda** [comment]
In my edge repository I have a build that does not have this issue. I can't explain why at this point, but this is why I wasn't able to reproduce. The automated workflow builds from the same source, it just publishes to a different endpoint (production vs dev).  I went over the build logs yesterday for both endpoints and they are identical.

1. Can I get one or two of you to verify that that is the case?
2. If it is the case that it works I will release a .1 update to 0.67 and hopefully it will fix the issue

Edge repo: https://github.com/haberda/hassio_addons_edge

If you get the edge working you can switch back to production when it is fixed by taking a backup and modifying the manifest file. That way you don't have to register numbers again.

**Comment by jprates** [comment]
"You just need to make sure that you restart the container afterwards. Otherwise your changes are reverted again."
You mean "(...) you don't restart (...)", right @bbernhard ?

@haberda  I'll try both your's and @bbernhard 's suggestions later on today or in the weekend at the latest.
Being an Oracle DBA myself with only basic linux know-how I might take some time to get there, I'm not a docker expert.

If anyone on this thread feels more comfortable  and can test ahead (and faster) then please go ahead and let us know how it went.

**Comment by bbernhard** [comment]
> You mean "(...) you don't restart (...)", right @bbernhard ?

ah, yeah, of course. Typo ;-)

**Comment by littlejogi** [comment]
@haberda, The edge version works fine on my home assistant server

**Comment by jprates** [comment]
YES!!!!!!!!!!!!
@haberda I can also report the edge version works fine on my HASS as well!

**Comment by haberda** [comment]
Ok I'll release a .1 update when I get home tonight. Hopefully that will fix it ¯\_(ツ)_/¯

**Comment by amroncz** [comment]
> Ok I'll release a .1 update when I get home tonight. Hopefully that will fix it ¯_(ツ)_/¯


I saw that you have already done it. Just tested it and it works. Thank you very much!

**Comment by jprates** [comment]
I'm really glad it is working for you all.

For me unfortunately it seems I got back to where I was before I tried to uninstall/deregister to solve the original problem:

I send messages and nothing happens.
In this case right now this means I send the registration curl command, and nothing happens, no 404 error anymore (good!) but also no feedback at all.
No SMS received on the phone number, no asking for captcha, nothing at all.
I'm going crazy with this.

Does anyone have a clue of what might be the root reason?
